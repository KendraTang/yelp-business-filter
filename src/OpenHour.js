import React from 'react'
import PropTypes from 'prop-types'
import moment from 'moment'

const OpenHour = ({day, hour}) => {
  const WEEKDAY_NAMES = '一二三四五六日'
  const add_colon = t => moment(t, 'HHmm').format('HH:mm')
  let open = hour.map(({ start, end }) => (
    start === '0000' && end === '0000' ?
      '全日營業' :
      `${add_colon(start)} ~ ${add_colon(end)}`
  )).join(', ')
  return (
    <div>
      星期{WEEKDAY_NAMES[day]}&emsp;{open}
    </div>
  )
}

OpenHour.propTypes = {
  day: PropTypes.number,
  hour: PropTypes.arrayOf(PropTypes.shape({
    start: PropTypes.string,
    end: PropTypes.string,
  })),
}

export default OpenHour
