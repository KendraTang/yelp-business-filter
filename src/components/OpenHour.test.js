import OpenHour from './OpenHour'

test('basic', () => {
  const props = {
    day: 0,
    hour: [{
      start: '1730',
      end: '2200',
    }]
  }
  expect(OpenHour(props)).toMatchSnapshot()
})

test('all day', () => {
  const props = {
    day: 0,
    hour: [{
      start: '0000',
      end: '0000',
    }]
  }
  expect(OpenHour(props)).toMatchSnapshot()
})

test('overnight', () => {
  const props = {
    day: 0,
    hour: [{
      start: '2300',
      end: '0200',
    }]
  }
  expect(OpenHour(props)).toMatchSnapshot()
})

test('weekday names', () => {
  for (let i = 0; i < 7; i++) {
    const props = {
      day: i,
      hour: [{
        start: '1100',
        end: '2100',
      }]
    }
    expect(OpenHour(props)).toMatchSnapshot()
  }
})

test('multiple hours in one day', () => {
  const props = {
    day: 0,
    hour: [
      {
        start: '0600',
        end: '0900',
      },
      {
        start: '1100',
        end: '1430',
      },
      {
        start: '1730',
        end: '2200',
      },
    ],
  }
  expect(OpenHour(props)).toMatchSnapshot()
})
