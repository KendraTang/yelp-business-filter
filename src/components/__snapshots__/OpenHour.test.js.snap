// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all day 1`] = `
<div>
  星期
  一
   
  全日營業
</div>
`;

exports[`basic 1`] = `
<div>
  星期
  一
   
  17:30 ~ 22:00
</div>
`;

exports[`multiple hours in one day 1`] = `
<div>
  星期
  一
   
  06:00 ~ 09:00, 11:00 ~ 14:30, 17:30 ~ 22:00
</div>
`;

exports[`overnight 1`] = `
<div>
  星期
  一
   
  23:00 ~ 02:00
</div>
`;

exports[`weekday names 1`] = `
<div>
  星期
  一
   
  11:00 ~ 21:00
</div>
`;

exports[`weekday names 2`] = `
<div>
  星期
  二
   
  11:00 ~ 21:00
</div>
`;

exports[`weekday names 3`] = `
<div>
  星期
  三
   
  11:00 ~ 21:00
</div>
`;

exports[`weekday names 4`] = `
<div>
  星期
  四
   
  11:00 ~ 21:00
</div>
`;

exports[`weekday names 5`] = `
<div>
  星期
  五
   
  11:00 ~ 21:00
</div>
`;

exports[`weekday names 6`] = `
<div>
  星期
  六
   
  11:00 ~ 21:00
</div>
`;

exports[`weekday names 7`] = `
<div>
  星期
  日
   
  11:00 ~ 21:00
</div>
`;
